runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -374,22 +383,28 @@ PCRE2_SIZE set_matches(const RegexInfo &regex_info, int64_t flags, mixed &matche
     output.push_back(output_val);
   }
 
-  matches = output;
-  return ovector[1];
+  (*opt_matches).get() = std::move(output);
+  return end_offset;
 }
 
 // returns the ending offset of the entire match
 // *** importrant ***
 // in case of a pattern order all_matches must already contain all groups as empty arrays before the first call to set_all_matches
-PCRE2_SIZE set_all_matches(const RegexInfo &regex_info, int64_t flags, mixed &all_matches) noexcept {
+PCRE2_SIZE set_all_matches(const RegexInfo &regex_info, int64_t flags, std::optional<std::reference_wrapper<mixed>> opt_all_matches) noexcept {
   const auto pattern_order{!static_cast<bool>(flags & kphp::regex::PREG_SET_ORDER)};
 
+  // early return in case we don't actually need to set matches
+  if (!opt_all_matches.has_value()) {
+    return set_matches(regex_info, flags, std::nullopt, pattern_order ? trailing_unmatch::include : trailing_unmatch::skip);
+  }
+
   mixed matches;
   PCRE2_SIZE offset{set_matches(regex_info, flags, matches, pattern_order ? trailing_unmatch::include : trailing_unmatch::skip)};
```

Comment:
```C++
PCRE2_SIZE offset{set_matches(regex_info, flags, matches, pattern_order ? trailing_unmatch::include : trailing_unmatch::skip)};
if (!opt_all_matches.has_value()) { return offset; }
mixed matches;
...
```

Reply:
It'd break the semantics of the function

Reply:
I don't see, give an example